Bottom: d2f05c24085f34fdfbaa9df21341239919af27ac
Top:    d5bfeed048ef8b563636f37bc5df9bbcc3ed192c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-14 13:01:43 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 819d4b5..c02e07d 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -227,6 +227,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
